### ISSUE TITLE
Improve spatial plugin layout and displayed widget names

### ISSUE THREAD

**Issue by constantinpape** [computational-cell-analytics/micro-sam#336]
The widgets added by the plugins now add up a lot of space, because the widget names are written next to the controls.
See screenshot below.

It would be better to put the names on top of the controls, so that less horizontal space is taken up.
We can also rethink the widget names that are displayed there.

![image](https://github.com/computational-cell-analytics/micro-sam/assets/4263537/e8d9c542-b60a-4d8f-b9e5-b86154c29128)

**Comment by GenevieveBuckley** [comment]
I looked into this, because it *feels* like it should be an easy change, but I wasn't able to figure it out. Things I found out:
* `layout="vertical"` is already the default setting
* There is a `labels=False` kwarg (in both `magic_factory` and `Container`) you can use to turn off the widget titles
* You can add a line of text as a label to the magic_factory decorated function, but you cannot put this at the very top unless there are no positional arguments (i.e. all function arguments are keyword arguments). That's often not the case.
```python
@magic_factory
def function_widget(
    image: "napari.layers.Image",
    ... # other arguments
    title: str = "WIDGET LABEL HERE",
)
    ... # function code
```

That feels like kind of a hacky workaround, I don't know if I like it.

**Comment by tlambert03** [comment]
there's no easy way to do this at the moment,  labels are always on the left in a form-like layout at the moment and there's no quick toggle to change that using `magicgui/factory`

**Comment by GenevieveBuckley** [comment]
Thanks @tlambert03 - it's good to know we weren't overlooking something obvious

**Comment by constantinpape** [comment]
I have addressed this by shortening the widget names in #409. This makes the layout more concise and solves this issue for now.

![image](https://github.com/computational-cell-analytics/micro-sam/assets/4263537/7fb5b073-91f7-4fbc-932f-8478692f3bc0)

**Comment by GenevieveBuckley** [comment]
That does look a lot better!
